include/nuttx/atomic.h

```diff
@@ -86,13 +86,149 @@ extern "C++"
 #    ifndef ATOMIC_VAR_INIT
 #      define ATOMIC_VAR_INIT(value) (value)
 #    endif
-#  else
-#    include <nuttx/lib/stdatomic.h>
 #  endif
-#else
-#  include <nuttx/lib/stdatomic.h>
 #endif
 
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef __ATOMIC_RELAXED
+#  define __ATOMIC_RELAXED 0
+#endif
+
+#ifndef __ATOMIC_CONSUME
+#  define __ATOMIC_CONSUME 1
+#endif
+
+#ifndef __ATOMIC_ACQUIRE
+#  define __ATOMIC_ACQUIRE 2
+#endif
+
+#ifndef __ATOMIC_RELEASE
+#  define __ATOMIC_RELEASE 3
+#endif
+
+#ifndef __ATOMIC_ACQ_REL
+#  define __ATOMIC_ACQ_REL 4
+#endif
+
+#ifndef __ATOMIC_SEQ_CST
+#  define __ATOMIC_SEQ_CST 5
+#endif
+
+#define atomic_set_explicit(obj, val, type) \
+  __atomic_store_4(obj, val, type)
+#define atomic_set(obj, val) \
+  atomic_set_explicit(obj, val, __ATOMIC_RELAXED)
+#define atomic_set_release(obj, val) \
+  atomic_set_explicit(obj, val, __ATOMIC_RELEASE)
+
+#define atomic_read_explicit(obj, type) \
+  (int)__atomic_load_4(obj, type)
```

Comment:
why need the cast in __atomic_load_4/__atomic_exchange_4/__atomic_flags_test_and_set_4...

Reply:
when use arch_atomic, these function return uint32_t type

Reply:
let's change the prototype to signed integer from:
https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html
compiler expect a signed integer (e.g. __int128).

Reply:
ok, let's use cast.